makefile

```diff
@@ -1,62 +1,55 @@
 #
-# Paths
+# OUTPUT PATHS
 #
-_QROOT_DIR = $(dir $(realpath $(lastword $(MAKEFILE_LIST))))
-QROOT_DIR = $(realpath $(_QROOT_DIR)../Quark)
-QKERNEL_BUILD_DIR = $(QROOT_DIR)/build
-QTARGET_RELASE = $(QROOT_DIR)/target/release
-QTARGET_DEBUG = $(QROOT_DIR)/target/debug
-QBIN_DIR ?= /usr/local/bin
-QCONFIG_GLOBAL_DIR ?= /etc/quark
+PREFIX          ?= /usr/local
+QBIN_DIR        ?= $(PREFIX)/bin
+QCONFIG_DIR     ?= /etc/quark
+QLOG_DIR        ?= /var/log/quark
 
 #
-# Bins
+# BUILD PATHS
 #
-QUARK = quark
-QUARK_DBG = quark_d
-QKERNEL_BIN_REL = qkernel.bin
-QKERNEL_BIN_DBG = qkernel_d.bin
-QKERNEL_DEBUG = $(QKERNEL_BUILD_DIR)/$(QKERNEL_BIN_DBG)
-QKERNEL_RELEASE = $(QKERNEL_BUILD_DIR)/$(QKERNEL_BIN_REL)
-QUARK_BIN_DEBUG = $(QTARGET_DEBUG)/$(QUARK)
-QUARK_BIN_RELEASE = $(QTARGET_RELASE)/$(QUARK)
+QKERNEL_BUILD   = build
+QTARGET_RELEASE = target/release
+QTARGET_DEBUG   = target/debug
+QKERNEL_DEBUG   = $(QKERNEL_BUILD)/qkernel_d.bin
+QKERNEL_RELEASE = $(QKERNEL_BUILD)/qkernel.bin
+QUARK_DEBUG     = $(QTARGET_DEBUG)/quark
+QUARK_RELEASE   = $(QTARGET_RELEASE)/quark
+VDSO            = vdso/vdso.so
 
-#
-# Flags
-#
-MAKEFLAGS += -j$(shell nproc)
-#
-# We do not support cross-compilation yet.
-#
 ARCH := ${shell uname -m}
-RUST_TOOLCHAIN = nightly-2023-12-11-$(ARCH)-unknown-linux-gnu
+RUST_TOOLCHAIN  = nightly-2023-12-11-$(ARCH)-unknown-linux-gnu
+
 
 .PHONY: all release debug clean install qvisor_release qvisor_debug cuda_make cuda_all
 
 all:: release debug
 
 cuda_all:: cuda_release cuda_debug
 
-release:: qvisor_release qkernel_release 
+release:: qvisor_release qkernel_release $(VDSO)
+
+debug:: qvisor_debug qkernel_debug vdso $(VDSO)
```

Comment:
Is the "vdso $(VDSO)" intentional?

Reply:
no, I'll remove it, thanks.

The idea was to explicitly require the vdso build target (`vdso/vdso.so`) so it only needs to be built once.